packages/core/lib/package.js

```diff
@@ -1,351 +1,214 @@
 const expect = require("@truffle/expect");
 const TruffleError = require("@truffle/error");
-const Networks = require("./networks");
-const EthPM = require("ethpm");
-const EthPMRegistry = require("ethpm-registry");
-const Web3 = require("web3");
-const { createInterfaceAdapter } = require("@truffle/interface-adapter");
-const async = require("async");
+const { EthPM } = require("ethpm");
+const { Erc1319URI } = require("ethpm/utils/uri");
+const { parseTruffleArtifacts } = require("ethpm/utils/truffle");
 const path = require("path");
 const fs = require("fs");
-const OS = require("os");
 
 const Package = {
-  install: async function(options, callback) {
-    expect.options(options, ["working_directory", "ethpm"]);
-
-    expect.options(options.ethpm, ["registry", "ipfs_host"]);
-
-    expect.one(options.ethpm, ["provider", "install_provider_uri"]);
-
-    // ipfs_port and ipfs_protocol are optinal.
+  install: async options => {
+    expect.options(options.ethpm, ["ipfs_host", "ipfs_port", "ipfs_protocol"]);
+    expect.options(options, ["working_directory", "ethpm_uri", "logger"]);
 
+    // Create a web3 instance connected to a blockchain
+    if (!options.provider && !options.ethpm.install_provider_uri) {
```

Comment:
I think we don't want to use the network provider, since there's no reason to believe that the user will be developing against the same network as the network they're using for ethpm